docs/src/tutorials/child-device-config-mngmnt.md

```diff
@@ -0,0 +1,297 @@
+
+
+
+Configuration management can be enabled for child devices using the same `c8y-configuration-plugin`,
+used for configuration management on thin-edge.io devices which you can read [here](https://thin-edge.github.io/thin-edge.io/html/howto-guides/025_config_management_plugin.html).
+
+A child device has to be able to handle the following responsibilities:
+
+* Declare the supported configuration list to thin-edge.io
+* Handle configuration snapshot requests from thin-edge.io
+* Handle configuration update requests from thin-edge.io
+
+**Please note:** The software that will handle the above responsibilities is not part of thin-edge.io. This software is referred to as `child device agent` in this document.
+
+The `supported configuration list` is the list of configuration files on the child device that needs to be managed from the cloud. Configuration management by thin-edge.io is enabled only for the files provided in this list. These declared configuration files can be fetched from thin-edge.io with config snapshot requests and can be updated with config update requests.
+
+Handling the above mentioned responsibilities involve multiple interactions with thin-edge.io:
+* via MQTT to receive and respond to configuration management requests.
+* via HTTP to upload/download files while handling those requests.
+
+For example, during the bootstrapping/startup of the child device, the child device agent needs to upload the supported configuration list of the child device to thin-edge.io by uploading a file using the HTTP file-transfer API of thin-edge.io, followed by an MQTT message informing thin-edge.io that the upload is completed. Similarly, handling a configuration snapshot or update request involves sending MQTT messages before and after the configuration file is uploaded/downloaded via HTTP to/from thin-edge.io.
+
+Since child device agents typically run on an external device and not on the thin-edge.io device itself, the MQTT and HTTP API's of thin-edge.io need to be accessed over the network using its IP address, which is configured using the thin-edge.io configuration settings `mqtt.external.bind_address` or `mqtt.bind_address`. The MQTT APIs are exposed via port 1883 and the HTTP API's are exposed via port 8000. In rare cases, where the child device agent is installed alongside thin-edge.io on the same device, these API's can be accessed via a local IP or even 127.0.0.1.
+
+In this tutorial 127.0.0.1. is used.
+
+##  Prerequisite
+
+To follow this tutorial, you only need the following:
+
+- A [Cumulocity IoT](https://www.softwareag.cloud/site/product/cumulocity-iot.html) tenant.
+- Installed the latest version of thin-edge.io. To this, you can follow [this tutorial](https://thin-edge.github.io/thin-edge.io/html/tutorials/getting-started.html#Step-6-Manage-configuration-files), which will also give you some deeper insights in the various capabilities of thin-edge.io, like configuration management.
```

Comment:
Install where? On the Raspberry Pi? In that case you need to put the Raspberry Pi as req. 2 and this as req. 3.